routes/api/contacts.js

```diff
@@ -1,25 +1,73 @@
-const express = require('express')
+const express = require("express");
+const fs = require("fs/promises");
+const {
+  listContacts,
+  getContactById,
+  updateContact,
+  removeContact,
+  addContact,
+} = require("../../models/contacts");
+const Joi = require("joi");
+var nanoid = require("nanoid-esm");
 
-const router = express.Router()
+const reqBodySchema = Joi.object({
+  name: Joi.string().min(3).max(30).required(),
+  email: Joi.string().email({ minDomainSegments: 2 }).required(),
+  phone: Joi.string().min(6).max(20).required(),
+});
+const router = express.Router();
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  const contacts = await listContacts();
+  res.json({ contacts });
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:id", async (req, res, next) => {
+  const { id } = req.params;
+  const searchedContact = await getContactById(id);
+  !searchedContact
+    ? res.status(404).json({ message: `There is no contacts with ID: ${id}` })
+    : res.json({ searchedContact });
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  const { value, error } = reqBodySchema.validate(req.body);
```

Comment:
Staraj się być bardziej precyzyjna: 
```suggestion
  const { value, error } = contactSchema.validate(req.body);
```